snapcraft/parts/plugins/matter_sdk_plugin.py

```diff
@@ -178,14 +167,8 @@ def get_build_commands(self) -> List[str]:
         # Compare and output pigweed related environment variables to matter_sdk_env env file
         commands.extend(
             [
-                "set -a",
-                'echo "PATH=$PATH" >> matter_sdk_env',
-                'env | grep "^PW_" >> matter_sdk_env',
-                'echo "VIRTUAL_ENV=$VIRTUAL_ENV" >> matter_sdk_env',
-                'echo "CIPD_CACHE_DIR=$CIPD_CACHE_DIR" >> matter_sdk_env',
-                "set +a",
-                "echo 'pigweed related environment variables differences exported",
-                "to matter_sdk_env file'",
+                'echo "export PATH=$PATH" >> matter_sdk_env',
```

Comment:
The `PATH` variable at this stage contains the following: `/root/parts/lighting-hello/build/matter/third_party/pigweed/repo/out/host/host_tools:/root/parts/lighting-hello/build/matter/.environment/pigweed-venv/bin:/root/parts/lighting-hello/build/matter/.environment/cipd/packages/arm/bin:/root/parts/lighting-hello/build/matter/.environment/cipd/packages/arm:/root/parts/lighting-hello/build/matter/.environment/cipd/packages/zap:/root/parts/lighting-hello/build/matter/.environment/cipd/packages/pigweed/bin:/root/parts/lighting-hello/build/matter/.environment/cipd/packages/pigweed:/root/parts/lighting-hello/build/matter/.environment/cipd:/snap/snapcraft/x1/libexec/snapcraft:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin`.

So, the following is prepended to `PATH` during bootstrapping: `/root/parts/lighting-hello/build/matter/third_party/pigweed/repo/out/host/host_tools:/root/parts/lighting-hello/build/matter/.environment/pigweed-venv/bin:/root/parts/lighting-hello/build/matter/.environment/cipd/packages/arm/bin:/root/parts/lighting-hello/build/matter/.environment/cipd/packages/arm:/root/parts/lighting-hello/build/matter/.environment/cipd/packages/zap:/root/parts/lighting-hello/build/matter/.environment/cipd/packages/pigweed/bin:/root/parts/lighting-hello/build/matter/.environment/cipd/packages/pigweed:/root/parts/lighting-hello/build/matter/.environment/cipd`.

Writing the whole PATH and letting the consumer export it could have side effects if the original PATH ends up containing other part-specific paths. Instead, you could prepend the Matter SDK additions to the PATH, e.g.: `'echo "export PATH=$MATTER_SDK_PATH:\$PATH" >> matter_sdk_env.sh'`

The content of the `matter_sdk_env.sh` should be as follows:
```bash
export PATH=/root/parts/lighting-hello/build/matter/third_party/pigweed/repo/out/host/host_tools:/root/parts/lighting-hello/build/matter/.environment/pigweed-venv/bin:/root/parts/lighting-hello/build/matter/.environment/cipd/packages/arm/bin:/root/parts/lighting-hello/build/matter/.environment/cipd/packages/arm:/root/parts/lighting-hello/build/matter/.environment/cipd/packages/zap:/root/parts/lighting-hello/build/matter/.environment/cipd/packages/pigweed/bin:/root/parts/lighting-hello/build/matter/.environment/cipd/packages/pigweed:/root/parts/lighting-hello/build/matter/.environment/cipd:$PATH
```

Reply:
Thanks for the suggestion, `MATTER_SDK_PATH` has been added in commit 9f5c229f351d9d279e904d504b5f215ece1e8036.